src/Promitor.Integrations.Sinks.Statsd/StatsdMetricSink.cs

```diff
@@ -31,12 +46,13 @@ public async Task ReportMetricAsync(string metricName, string metricDescription,
             Guard.NotNull(scrapeResult.MetricValues, nameof(scrapeResult.MetricValues));
 
             var reportMetricTasks = new List<Task>();
+            var formatterType = _statsDConfiguration.CurrentValue?.FormatterType ?? StatsdFormatterTypes.DEFAULT;
 
             foreach (var measuredMetric in scrapeResult.MetricValues)
             {
                 var metricValue = measuredMetric.Value ?? 0;
 
-                var reportMetricTask = ReportMetricAsync(metricName, metricDescription, metricValue, new Dictionary<string, string>());
+                var reportMetricTask = _reportMetricsActions[formatterType](metricName, metricDescription, metricValue, scrapeResult.Labels);
```

Comment:
Can't we just register a tag formatter? Did you consider this?
https://github.com/justeat/JustEat.StatsD#tagging-support

Reply:
I checked, seems like I would have to implamant the exact same logic in the interface, maybe we can so it on stage 2?

Reply:
Sure, fine for me